service/src/main/java/tennisclub/service/UserServiceImpl.java

```diff
@@ -39,17 +43,51 @@ public User register(User user, String plainTextPassword) {
         return userDao.findAll();
     }
 
+    public String authenticateJWT(String username, String plainTextPassword) {
+        User user = findUserOrThrowUnauthorised(username);
+
+        if (passwordEncoder.matches(plainTextPassword, user.getPasswordHash())) {
+            return JWTService.createJWT(user);
```

Comment:
An instance of `JWTService` should probably be `@Autowired` and contain instance -- rather than static -- methods.

Reply:
Other than that, let me pull this branch and manually test the REST endpoint.